python/main.py

```diff
@@ -216,6 +222,10 @@ def create_mesh(header, modules_only=False, show_det_eff=False, random_color=Fal
 
                 shapes.append(module_mesh)
 
+            module_count += 1
+            if num_modules >= 0 and module_count > (num_modules * (skip_modules + 1)):
```

Comment:
```suggestion
            if num_modules > 0 and module_count + skip_modules + 1 > (num_modules * (skip_modules + 1)):
```
Left part of the "and": with the "=", not using the option would show only one module
Right part of the "and": Without that, it always show more than `num_modules`